core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/DefaultAwsCrtV4aHttpSigner.java

```diff
@@ -174,7 +174,11 @@ private static boolean isPayloadSigning(BaseSignRequest<?, ? extends AwsCredenti
         boolean isPayloadSigningEnabled = request.requireProperty(PAYLOAD_SIGNING_ENABLED, true);
         boolean isEncrypted = "https".equals(request.request().protocol());
 
-        return !isAnonymous && (isPayloadSigningEnabled || !isEncrypted);
+        if (isPayloadSigningEnabled) {
+            return !isAnonymous;
+        }
+
+        return !isAnonymous && !isEncrypted;
```

Comment:
These look like the same conditions. Under what conditions would they return different results before and after this change?